src/tests/fixing.rs

```diff
@@ -0,0 +1,159 @@
+#![cfg(test)]
+
+use std::sync::Arc;
+
+use tree_sitter::Node;
+use tree_sitter_grep::SupportedLanguage;
+
+use crate::{
+    context::QueryMatchContext,
+    rule::{FileRunInfo, Rule, RuleInstance, RuleInstancePerFile, RuleListenerQuery, RuleMeta},
+    run_fixing_for_slice, Config, ConfigBuilder, ViolationBuilder,
+};
+
+#[test]
+fn test_single_fix() {
+    let mut file_contents = r#"
```

Comment:
Ok there's a lot of test-case ergonomics that could be added here but this is doing an in-memory test of basic fixing behavior (that's basically specifying "input", "fixed output" and "rule")

Reply:
Yeah, it'd be nice to have helpers for this, but it's nice that it's so easily tested as input -> output like this.